src/js/components/stock-movement-wizard/outbound/AddItemsPage.jsx

```diff
@@ -729,6 +769,12 @@ class AddItemsPage extends Component {
     };
 
     if (payload.lineItems.length) {
+      this.props.addLines(
+        this.state.workflow,
+        itemCandidatesToSave,
+        this.state.values.stockMovementId,
+      );
+
```

Comment:
and here respecitvely:
```javascript
 this.props.addLines({
   workflow: this.state.workflow,
   lines: itemCandidatesToSave,
   id: this.state.values.stockMovementId,
 })

Reply:
By the way - don't we want to store those only if we lose connection? My visualization of this ticket was that you'd do something like this (pseudocode):
```javascript
  useEffect(() => {
    if (connection === offline) {
      create a draft (addLines to the redux)
    }
  }, [connectionStatus])
```

For now I can see that you add it to the reducer each time you save.
Obviously I might be wrong what the expectations were, but at the moment I can see that you do a lot of unnecessary redux operations even if you have connection.

Reply:
We have to store each change after losing connection, so you are right that there is a lot of unnecessary redux operations. I have to add an `if` statement before every adding lines to draft to reduce those operations. This approach with `useEffect` won't work because those lines have to be added to state in `then()`, `finally()`, etc. when we don't have internet connection, not when we are losing connection.

Reply:
Right, you can only change / add quantity, because other actions needs api calls (I don't remember if changing recipient needs an api call). If you edit quantity being offline the request is not sent, so it triggers the catch function and a row has `RowSaveStatus.ERROR` and then we are saving our lines as a draft version.